### PR TITLE
Updated new requests form component

### PR DESCRIPTION
**Pull request by DanielBaxan**
Updated the register new request form according to figma.

Since the new request-details-component represents the form itself, the form component was deleted from shared.

This component is currently used for registering a new request, and in future also for updating an existing one.

Current functionality for finding an existent beneficiary works as follows: If a beneficiary is found by mobile phone - an according div will show up, when clicking on this div the values in the form are updated accordingly.

After this update I would like to continue improving the existent beneficiary logic according to issue #153, and after finishing #153 I plan to work on the editing an existing request part of this form.

Thank you, please confirm the possibility to work on issue #153.

### CODE REVIEW

File: src/app/admin/requests/request-details/request-address-field/request-address-field.component.ts [hunk 1]
```diff
@@ -33,37 +35,35 @@ export class RequestAddressFieldComponent implements OnInit {
           coors: [47.02486150651041, 28.832740004203416],
           address: 'Arcul de Triumf',
         },
-        panelClass: 'cdk-overlay-pane-no-padding',
+        panelClass: 'esri-map',
```
Comment (unrevised6419) [inline src/app/admin/requests/request-details/request-address-field/request-address-field.component.ts]: What is the reason of changing this class?
This class removes the padding of the dialog. If the invention was to have dialog padding, that this `panelClass` can be removed at all

File: src/app/admin/requests/request-details/request-details.component.ts [hunk 2]
```diff
@@ -89,7 +183,22 @@ export class RequestDetailsComponent implements OnInit, OnDestroy {
...
-    this.form.get('address').patchValue(event.address);
+    this.form.get('beneficiary.address').patchValue(event.address);
     this.validAddress = event.valid;
   }
+
+  isEmpty(obj: Beneficiary) {
+    for (const key in obj) {
+      if (obj.hasOwnProperty(key)) return false;
+    }
+    return true;
```
Comment (unrevised6419) [inline src/app/admin/requests/request-details/request-details.component.ts]:
```suggestion
    return Object.keys(obj).length < 1;
```

**Review by unrevised6419 — CHANGES_REQUESTED** [review]

File: src/app/admin/requests/request-details/request-details.component.ts [hunk 3]
```diff
@@ -1,85 +1,179 @@
...
   templateUrl: './request-details.component.html',
   styleUrls: ['./request-details.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class RequestDetailsComponent implements OnInit, OnDestroy {
+export class RequestDetailsComponent implements OnInit {
   form: FormGroup;
   zones: Array<string> = Object.keys(ZONES).filter((key) => isNaN(+key));
-  needs = RequestTypeUpdated;
+  needs = DemandType;
```
Comment (unrevised6419) [inline src/app/admin/requests/request-details/request-details.component.ts]: Export from file `demand.ts` this, and use it here
```suggestion
  needs = demandTypes;
```
```ts
export const demandTypes = Object.values(DemandType);
```

File: src/app/admin/requests/request-details/request-details.component.html [hunk 4]
```diff
@@ -203,7 +225,7 @@
               {{ need.value }}
```
Comment (unrevised6419) [inline src/app/admin/requests/request-details/request-details.component.html]: If you made the changes with `export const demandTypes`
In `ngFor` you remove ` | keyvalue: enumUnsorted`, And use need as `[value]`
and also need for the title, but with translate, and add new translations to `ro.json` file
In the end you should have something like this

```html
<mat-option *ngFor="let need of needs" [value]="need">
  <mat-icon [svgIcon]="need"></mat-icon>
  {{ need | translate }}
</mat-option>
```

To use translate you will have to import `TranslateModule` in `RequestsModule`

**Review by unrevised6419 — CHANGES_REQUESTED** [review]

File: src/app/admin/requests/request-details/request-address-field/request-address-field.component.ts (hunk 1 above)
Comment (DanielBaxan) [reply src/app/admin/requests/request-details/request-address-field/request-address-field.component.ts]: Reverted back to cdk-overlay-pane-no-padding.

File: src/app/admin/requests/request-details/request-details.component.ts (hunk 2 above)
Comment (DanielBaxan) [reply src/app/admin/requests/request-details/request-details.component.ts]: Done

File: src/app/admin/requests/request-details/request-details.component.ts (hunk 3 above)
Comment (DanielBaxan) [reply src/app/admin/requests/request-details/request-details.component.ts]: Done

File: src/app/admin/requests/request-details/request-details.component.html (hunk 4 above)
Comment (DanielBaxan) [reply src/app/admin/requests/request-details/request-details.component.html]: Done

**Review by unrevised6419 — APPROVED** [review]
LGTM, but still can be improved

**Pull request merged**